web/src/invite_user_group_picker_pill.ts

```diff
@@ -43,9 +51,24 @@ export function get_user_groups_allowed_to_add_members(): UserGroup[] {
     return all_user_groups.filter((group) => settings_data.can_add_members_to_user_group(group.id));
 }
 
-function set_up_pill_typeahead({pill_widget, $pill_container}: SetUpPillTypeaheadConfig): void {
+export function get_user_groups_where_user_can_be_added(user_id: number): UserGroup[] {
+    const all_user_groups: UserGroup[] = user_groups.get_realm_user_groups();
+    user_groups_of_user = user_groups.get_user_groups_of_user(user_id);
```

Comment:
This will also include groups which the user is not a direct member of but is member of any of its subgroup and thus will not be shown in the typeahead. But we want to show them in the typeahead, because I guess we would still want to allow user to be added to such groups.